src/server/scripts/Spells/spell_dk.cpp

```diff
@@ -1689,6 +1655,73 @@ class spell_dk_death_grip_initial : public SpellScriptLoader
     }
 };
 
+// 49560 - Death Grip (second part)
+class spell_dk_death_grip : public SpellScriptLoader
+{
+public:
+    spell_dk_death_grip() : SpellScriptLoader("spell_dk_death_grip") { }
+
+    class spell_dk_death_grip_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_dk_death_grip_SpellScript);
+
+        void HandleDummy(SpellEffIndex /*effIndex*/)
+        {
+            int32 damage = GetEffectValue();
+            Position const* pos = GetExplTargetDest();
+            if (Unit* target = GetHitUnit())
+            {
+                if (!target->HasAuraType(SPELL_AURA_DEFLECT_SPELLS)) // Deterrence
+                    target->CastSpell(pos->GetPositionX(), pos->GetPositionY(), pos->GetPositionZ(), damage, true);
+            }
+        }
+
+        void Register() override
+        {
+            OnEffectHitTarget += SpellEffectFn(spell_dk_death_grip_SpellScript::HandleDummy, EFFECT_0, SPELL_EFFECT_DUMMY);
+        }
+
+    };
+
+    SpellScript* GetSpellScript() const override
+    {
+        return new spell_dk_death_grip_SpellScript();
+    }
+};
+
+// 49575 - Death Grip (third part. pull effect only)
+class spell_dk_death_grip_pull : public SpellScriptLoader
+{
+public:
+    spell_dk_death_grip_pull() : SpellScriptLoader("spell_dk_death_grip_pull") { }
+
+    class spell_dk_death_grip_pull_SpellScript : public SpellScript
+    {
+        PrepareSpellScript(spell_dk_death_grip_pull_SpellScript);
+
+        SpellCastResult CheckCast()
+        {
+            Creature* creatureTarget = GetCaster()->ToCreature(); // the caster is the unit affected by Death Grip (see spell script of previous spell)
+            if (creatureTarget && creatureTarget->GetCreatureTemplate()->flags_extra & CREATURE_FLAG_EXTRA_NO_PULL_BY_GRIP)
```

Comment:
Shouldn't this be checked in a more generic place and not in a very specific spellscript?
Doing it this way means that all spells with this mechanic need to be scripted individually. The flag should be checked for any spell with the pull effect.